nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -306,6 +310,15 @@ def build_pretraining_data_loader(self, dataset, consumed_samples):
         )
 
     def setup(self, stage=None):
+        resume_checkpoint_path = self.trainer.checkpoint_connector.resume_from_checkpoint_fit_path
+        if resume_checkpoint_path:
+            init_consumed_samples = int(
+                float(re.findall(r"consumed_samples\=([0-9]+.[0-9]+)", resume_checkpoint_path)[0])
```

Comment:
why int(float(...)) ?

Reply:
Nicer to have int on WANDB

Reply:
I forget why we had float there but I think it was needed

Reply:
The checkpoint filename for some reason has the format of  `consumed_samples=\d+.\d` , where \d are numbers 0-9. why we need a `.0` in the end?